src/test/java/uk/nhs/ctp/transform/ConditionDTOTransformerTest.java

```diff
@@ -0,0 +1,126 @@
+package uk.nhs.ctp.transform;
+
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+import java.util.Date;
+import org.hamcrest.Matchers;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.hl7.fhir.dstu3.model.Condition.ConditionClinicalStatus;
+import org.hl7.fhir.dstu3.model.Condition.ConditionEvidenceComponent;
+import org.hl7.fhir.dstu3.model.Condition.ConditionVerificationStatus;
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.IdType;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import uk.nhs.ctp.service.dto.ConditionDTO;
+import uk.nhs.ctp.service.fhir.GenericResourceLocator;
+import uk.nhs.ctp.service.fhir.ReferenceService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConditionDTOTransformerTest {
+
+  @Mock
+  GenericResourceLocator genericResourceLocator;
+
+  @Mock
+  ReferenceService referenceService;
+
+  @Test
+  public void dtoIncludesCodeElements() {
+    CodeableConcept code = new CodeableConcept()
+        .addCoding(new Coding()
+            .setSystem("system")
+            .setCode("code")
+        );
+
+    Observation observation = new Observation().setCode(code);
+    when(genericResourceLocator.findResource(any(Reference.class), any(IdType.class)))
```

Comment:
If you move this below setting up the condition you can make sure this is called with the correct reference and idtype rather than just 'any'

Reply:
I originally wrote this with eq() matchers, but it wasn't working for some reason. Would you suggest just using the objects I'm expecting without matchers in this case?

Reply:
Yes, in this case you could save those variables first and then make sure it's the same instance. You can also use `argThat(Matcher<T>)` and then provide a hamcrest matcher (e.g anything from `FhirMatchers`) if you don't have the exact instance.